src/shims/unix/fs.rs

```diff
@@ -273,7 +273,7 @@ impl UnixFileDescription for FileHandle {
 
 impl<'tcx> EvalContextExtPrivate<'tcx> for crate::MiriInterpCx<'tcx> {}
 trait EvalContextExtPrivate<'tcx>: crate::MiriInterpCxExt<'tcx> {
-    fn macos_fbsd_solaris_write_buf(
+    fn macos_fbsd_solarish_write_buf(
```

Comment:
If we're renaming anyway, what about `write_stat` or so? "buf" is very non-descriptive.